### PR TITLE
1350 Fix PD/DQ/DR piping

### PR DESCRIPTION
**Pull request by leite08**
Ticket: https://github.com/metriport/metriport-internal/issues/1350

### Dependencies

- Upstream
   - https://github.com/metriport/metriport/pull/1591
   - https://github.com/metriport/metriport-internal/pull/1548
- Downstream: none

### Description

- Rename/split IHE GW outbound result functions 
- Local implementation (factory) for CQ DQ/DR IHE GW result polling
- Fix PD resp. from IHE GW, `externalGatewayPatient`

### Testing

- Local
  - [x] PD works
  - [x] DQ works
  - [x] DR gets initiated (didn't work fully)

### Release Plan

- [x] Merge upstream
- [ ] Merge this
- [ ] Update IHE GW config on `staging`

### CODE REVIEW

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA DQ ITI-38 Bulk Interface.xml [hunk 1]
```diff
@@ -258,7 +258,7 @@ Last updated: Dec 20 2023</description>
   </properties>
   <exportData>
     <metadata>
-      <enabled>false</enabled>
+      <enabled>true</enabled>
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA DQ ITI-38 Bulk Interface.xml:261]: re-enable channel

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCPD Bulk Interface.xml [hunk 2]
```diff
@@ -250,7 +250,7 @@ Last updated: Dec 08 2023</description>
   </properties>
   <exportData>
     <metadata>
-      <enabled>false</enabled>
+      <enabled>true</enabled>
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCPD Bulk Interface.xml:253]: re-enable channel

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA DR ITI-39 Bulk Interface.xml [hunk 3]
```diff
@@ -253,7 +253,7 @@ Last updated: Dec 05 2023</description>
   </properties>
   <exportData>
     <metadata>
-      <enabled>false</enabled>
+      <enabled>true</enabled>
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA DR ITI-39 Bulk Interface.xml:256]: re-enable channel

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA - DR App Interface.xml [hunk 4]
```diff
@@ -1,485 +1,243 @@
...
-        
         <username/>
-        
         <password/>
-        
         <content>${message.encodedData}</content>
-        
-        <contentType>text/plain</contentType>
-        
+        <contentType>application/json</contentType>
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA - DR App Interface.xml:128]: update content type

File: packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA - DQ App Interface.xml [hunk 5]
```diff
@@ -1,487 +1,245 @@
...
-        
         <username/>
-        
         <password/>
-        
         <content>${message.encodedData}</content>
-        
-        <contentType>text/plain</contentType>
-        
+        <contentType>application/json</contentType>
```
Comment (leite08) [inline packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA - DQ App Interface.xml:130]: update content type

File: packages/infra/lib/ihe-stack/ihe-gateway.ts [hunk 6]
```diff
@@ -76,7 +75,7 @@ export function createIHEGateway(stack: Construct, props: IHEGatewayProps): void
     db,
     name: `${name}Inbound`,
     dnsSubdomain: "inbound",
-    httpPorts: [portInboundPD, portInboundDQ],
+    httpPorts: [config.inboundPorts.patientDiscovery, config.inboundPorts.documentQuery],
```
Comment (Goncharo) [inline packages/infra/lib/ihe-stack/ihe-gateway.ts:78]: why is inbound DR not a thing?

**Review by Goncharo — APPROVED** [review]
code LGTM - one Q about inbount DR

File: packages/infra/lib/ihe-stack/ihe-gateway.ts (hunk 6 above)
Comment (leite08) [reply packages/infra/lib/ihe-stack/ihe-gateway.ts:78]: both use the same channel, `XCA Inbound Interface`

**Review by Orta21 — APPROVED** [review]

**Pull request merged**